src/main/java/com/xinecraft/listeners/PlayerJoinLeaveListener.java

```diff
@@ -32,6 +44,36 @@ public void onPlayerJoin(PlayerJoinEvent event) {
 
         // perform whois & add player to list of linkedPlayers hashmap
         this.broadcastWhoisForPlayer(event.getPlayer());
+
+        // fireworks effect when player joins
+        if (Minetrax.getPlugin().getIsFireworkOnPlayerJoin()) {
+            Firework fw = player.getWorld().spawn(player.getLocation(), Firework.class);
```

Comment:
Code duplications when firing fireworks.
Can be changed to function.

Reply:
Will work on this later today